lib/changeNIDMtoxSPM.m

```diff
@@ -1,57 +1,70 @@
+%==========================================================================
+%Generate an object with the same format as the SPM-output variable, xSPM, 
+%using the information from an input NIDM-Results json pack. This takes in 
+%one argument:
+%
+%json - the spm_jsonread form of the NIDM-Results json file.
+%
+%Authors: Thomas Maullin, Camille Maumet.
+%==========================================================================
+
 function NxSPM = changeNIDMtoxSPM(json)
     
     graph = json.('x_graph');
     NxSPM = struct;
     
-    %==============================================
+    %======================================================================
     %title
     
+    %Find the contrast map holding the title.
     contrastMaps = searchforType('nidm_ContrastMap', graph);
     for i = 1:length(contrastMaps)
         if isfield(contrastMaps{i}, 'nidm_contrastName')
             titleTemp = contrastMaps{i}.('nidm_contrastName');
         end
     end 
     
-    %==============================================
+    %======================================================================
     %STAT
     
-    statisticMaps = searchforType('nidm_StatisticMap', graph);
-    for i = 1:length(statisticMaps)
-        if isfield(statisticMaps{i}, 'nidm_statisticType')
-            statType = statisticMaps{i}.('nidm_statisticType').('x_id');
-        end
-    end
-    
-    if strcmp(statType, 'obo:STATO_0000176')
-       STATTemp = 'T';
-    elseif strcmp(statType, 'obo:STATO_0000030')
-       STATTemp = 'X';
-    elseif strcmp(statType, 'obo:STATO_0000378')
-       STATTemp = 'Z';
-    elseif strcmp(statType, 'obo:STATO_0000282')
-       STATTemp = 'F';
-    else
-       STATTemp = 'P';
-    end
+    %Using getStatType, obtain the statisticMaps objects and statistic type. 
+    [STATTemp, statisticMaps] = getStatType(graph);
        
-    %===============================================
+    %======================================================================
     %STATStr
     
+    %Obtain the degrees of freedom.
     for i = 1:length(statisticMaps)
         if isfield(statisticMaps{i}, 'nidm_errorDegreesOfFreedom')
-            errorDegrees = statisticMaps{i}.('nidm_errorDegreesOfFreedom').('x_value');
+            anyStatType = statisticMaps{i}.('nidm_statisticType').('x_id');
+            if ~strcmp(anyStatType, 'obo:STATO_0000376')
+                effectDegrees = statisticMaps{i}.('nidm_effectDegreesOfFreedom').('x_value');
```

Comment:
Would it be okay if this is addressed in the next pull request where I aim to have the viewer working for multiple contrasts?